internal/pkg/template/template.go

```diff
@@ -63,6 +100,56 @@ func (t *Template) Parse(path string, data interface{}, options ...ParseOption)
 	return &Content{buf}, nil
 }
 
+// UploadEnvironmentCustomResources uploads the environment custom resource scripts.
+func (t *Template) UploadEnvironmentCustomResources(upload s3.UploadFunc) ([]string, error) {
+	return t.uploadCustomResources(upload, envCustomResourceFiles)
+}
+
+func (t *Template) uploadCustomResources(upload s3.UploadFunc, fileNames []string) ([]string, error) {
+	var urls []string
+	for _, name := range fileNames {
+		url, err := t.uploadCompressedFile(upload, compressedFile{
+			name: path.Join(scriptDirName, name),
+			uploadables: []Uploadable{
+				{
+					name: customResourceZippedScriptName,
+					path: path.Join(customResourceRootPath, fmt.Sprintf("%s.js", name)),
+				},
+			},
+		})
+		if err != nil {
+			return nil, err
+		}
+		urls = append(urls, url)
+	}
+	return urls, nil
+}
+
+func (t *Template) uploadCompressedFile(upload s3.UploadFunc, file compressedFile) (string, error) {
```

Comment:
A small question: does it matter if the file will be compressed inside this function? 

I ask this because the files are only compressed inside the `upload` function that is passed from `env init`. So I am not sure if "compress" is the best word to describe this function. If we ignore the function name, it doesn't seem like anything is being compressed here.

In fact I am not sure this is more of a question regarding the naming or implementation, i.e. whether we should simply change the function name or require that `upload` parameter is of type, say, `CompressAndUploadFunc`.

However I might be missing some important assumptions here so I might be asking the wrong question 😮

Reply:
good call! Updated.